notes/NewSiteArchitecture.md

```diff
@@ -0,0 +1,174 @@
+* Possibilities are endless for connecting to your server's existing data model.
+
+Finally, there will be a set of links for learning more (diving into `Learn`) and for getting started (in `Code`). As well as a wall-o-logo for companies using GraphQL.
+
+## Learn
+
+*Goal:* Introduce GraphQL, one concept at a time, covering both primary concepts and best practices.
+
+*Timeframe:* Basic primary concepts by Sept 12th, advanced primary concepts by Sept 30th, best practices as ready over Q3/Q4.
+
+Where "GraphQL the Spec" is designed for a specific audience of those building GraphQL servers, this represents "GraphQL the Book" and is designed for the audience of anyone who wishes to use GraphQL. It should cover both GraphQL core concepts in addition to best practices and further topics, and it should range from introductory concepts through advanced concepts.
+
+The landing page for this section should begin as a more information-rich introduction to GraphQL, explaining why you might use it, and give a brief overview of the constituent parts. Take-aways from this introduction page:
+
+* GraphQL is a query language.
+* GraphQL servers describe a type system, called a "schema".
+* Clients can access a GraphQL server's type system to learn about what's possible.
+* Clients send queries to servers and typically get back JSON.
+* GraphQL servers validate and execute GraphQL queries.
+
+There is then a TOC through the sections and chapters (this is a straw-man list, open to reordering and addition)
+
+* Core Concepts:
+  * Query Language:
+    * Basics (queries, fields, arguments, aliases, comments)
+    * Variables
+    * Fragments
+    * Values
+    * Directives (skip & include)
+  * Type System:
+    * Basics (Schema, Objects & Fields)
+    * Scalars
+    * Lists & NonNull
+    * Interfaces
+    * Unions
+    * Enums
+  * Advanced Topics:
+    * Validation
+    * Execution & Error Handling
+    * Introspection
+    * Mutations
+* Best Practices:
+  * Servers:
+    * Schema Design Guidelines
+    * Paginating Lists
+    * Non-Breaking Changes
+    * Performant Servers (Batching & Caching)
+    * Authentication & Authorization
+    * Serving over HTTP
+  * Clients:
+    * Using Variables
+    * Co-locating Fragments
+    * Caching Results
+    * Persisted Queries
+    * Generating Models
+
+## Code
+
+*Goal:* Introduce open source GraphQL tools along with quick getting started guidelines for each.
+
+*Timeframe:* Top 3 servers described by Sept 12th, remainder by Sept 30th.
+
+This page is all about fulfilling the "Ok I'm sold! Now what?" conundrum. It should first very quickly reintroduce the elements of GraphQL you would expect to see software for as well as offer a quick path towards getting something working.
+
+1) Servers
+
+Explain the purpose of a GraphQL server, that there are servers written for many different languages and environments, and that graphql-js is the reference implementation operated by Facebook.
```

Comment:
I think I would agree with @lacker and @stubailo on this one. On one hand, it would be nice not to  give any particular implementation "premium" status since it can be discouraging, but on the other hand it always helpful and encouraging to immediately see something real: real code that does something useful and gives the reader feeling: "This looks simple enough, I can do it too!".

The thing is, the landing page and some other pages contain a lot of abstract concepts and many of these concepts would be new to the reader. Giving too much theory in one big chunk may cause fatigue, even if these concepts are very interesting. That's why I always in favor of balancing theory and abstract concepts with something very concrete and down-to-earth. This this case concrete implementation of these concepts could be the javascript/java/ruby/scala/etc... code.

I think maybe we can get some inspiration from [MessagePack](http://msgpack.org) landing page. When I first saw this page for the first time, it immediately gave impression that:

* It is a generic data format
* It is supported by may programming languages which are treated equally
* I can see something concrete when I click on particular language and evaluate it's nature/how it may fit in my application

One issue with MessagePack page is lack of polish, in my opinion. I I think this raises a bigger question: if we include some examples in different languages then ideally they should:

* represent the same schema/fields/concept or in other words, conceptually they should demonstrate the same thing
* they should be maintained over the time as libraries grow

I really like what @steveluscher did with [zero-to-graphql](https://github.com/steveluscher/zero-to-graphql). It provides implementation of the same GraphQL schema in different languages. All of these examples are implementing very similar application with same architecture - it's a web application that serves JSON over HTTP and uses an SQL database to store the data. The schema itself is very minimal but still able to demonstrate a lot of concepts (like many-to-many relationships).

Maybe we can expand on this idea and make it "official"/supported part of the ecosystem? If there is interest of library maintainers and the schema/application is small enough to make it easy to maintain, then maybe it can work. I feel that at this point it does not matter whether it is embedded in the graphql.org or maintained somewhere externally (e.g. on the github as a monorepo or a a set of independent repositories). It's more the learning experience and some quality/uniformity standard for these examples that do matter.